README.md

```diff
@@ -15,21 +15,135 @@ ready until after the talk.
+  }
+  object.set('keywords_list', object.get('keywords').split(' '));
+  response.success(object);
+}
+```
+
+A logical structure for your blog posts in Firebase might look like:
+
+```json
+{
+  "posts": {
+    "aBlogPostID": {
+      "title": "Firebase expands to become a unified app platform",
+      "body": "Eighteen months ago, Firebase joined Google...",
+      "keywords": ["googleio", "announcements"],
+    },
+  },
+  "keywords": {
+    "googleio": {
+      "aBlogPostID": true,
+      "anotherBlogAboutGoogleIO": true
+    },
+    "announcements": {
+      "aBlogPostID": true
+    }
+  }
+}
+```
+
+A few things are noticeably different in this structure:
+
+1. We no longer use the legacy `"keywords"` array
+2. The values from `"keywords_list"` are stored in a different location so we can look up blogs related to a `"googleio"` by reading the keys under `"/keywords/googleio"`
+
+Migrator can help us do this very easily in 3 steps:
+
+1. Change our Parse.beforeSave to a Migrator.beforeSave. This is the same except it works with promises instead of callbacks.
+2. Describe how we would copy an object from Parse to Firebase
+3. (optional) Describe how we translate a Parse object deletion to Fireabse
+
+We would do this with the following Cloud Code:
+
+```js
+var _ = require('underscore');
+var migrator = require('cloud/migrator.js')(Parse);
+var ref = require('cloud/firebase.js')({
+  url: 'https://my-firebase-database-url.firebaseio.com',
+  key: 'if-my-firebase-database-is-secured-this-key-gives-me-admin-access',
+  parse: Parse
+});
+
+// Notice this is almost the same (and actually a bit simpler) than the original beforeSave
+migrator.beforeSave('BlogPost', function(request) {
+  var object = request.object;
+  if (object.isDirty('keywords_list')) {
+    throw 'keywords_list is managed by Cloud Code only';
+  }
+  object.set('keywords_list', object.get('keywords').split(' '));
+  return object;
+}
+
+// When an object is created, this is run just after the afterSave (we need a second pass
+// to get an object ID). When the object is modified, this is run between beforeSave and
+// afterSave.
+migrator.migrateObject('BlogPost', function(object) {
+  // We're setting the full object
+  var saveObject = ref.child('posts').child(object.id).put({
+    title: object.get('title'),
+    body: object.get('body'),
+    keywords: object.get('keywords')
+  });
+  
+  // But using a deep update to only adjust keywords
+  var keywordsUpdate = {};
+  _.each(object.get('keywords'), function(keyword) {
+    keywordsUpdate['keywords/' + keyword + '/' + object.id] = true;
+  });
+  
+  return Parse.Promise.when(saveObject, ref.patch(keywordsUpdate));
+});
+
+// This is run between beforeDelete and afterDelete.
+migrator.migrateDelete('BlogPost', function(object) {
+  // Setting a value to null in Firebase deletes it
+  var updates = {};
+  _.each(object.get('keywords'), function(keyword) {
+    updates['keywords/' + keyword + '/' + object.id] = null;
```

Comment:
Ah, I would note above when you put "keywords" in your firebase schema that it exists solely to make deleting objects more efficient.